core/crypto/hasher.hpp

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_HASHER_HASHER_HPP_
+#define KAGOME_CORE_HASHER_HASHER_HPP_
+
+#include "common/blob.hpp"
+#include "common/buffer.hpp"
+
+namespace kagome::hash {
+  class Hasher {
+    using Hash128 = kagome::common::Hash128;
+    using Hash256 = kagome::common::Hash256;
+    using Buffer = kagome::common::Buffer;
+
+   public:
+    virtual ~Hasher() = default;
+
+    /**
+     * @brief hashTwox128 calculates 16-byte twox hash
+     * @param buffer source buffer
+     * @return 128-bit hash value
+     */
+    virtual Hash128 hashTwox_128(const Buffer &buffer) = 0;
```

Comment:
why do you use prefix `hash`? 
if object name is Hasher, IMO it is obvious that it calculates hashes :)

Reply:
It seems to me, that some prefix like hash or make is desirable.
Without it names will be lime Hasher::twox_128;
Hasher hasher{};
hasher.twox_128(buffer);
Not suitable in my opinion.

Reply:
renamed functions